src/app/dynamic/enterprise/metering/schedule-config/report-list/component.ts

```diff
@@ -19,77 +19,85 @@
+import {Subject} from 'rxjs';
+import {takeUntil} from 'rxjs/operators';
+import {MeteringService} from '@app/dynamic/enterprise/metering/service/metering';
 import {UserService} from '@core/services/user';
-import {MeteringConfiguration} from '@shared/entity/datacenter';
 import {Report} from '@shared/entity/metering';
-import {merge, of, Subject} from 'rxjs';
-import {filter, switchMap, takeUntil} from 'rxjs/operators';
 
 @Component({
-  selector: 'km-metering-list',
+  selector: 'km-metering-reports-list',
   templateUrl: './template.html',
+  styleUrls: ['style.scss'],
 })
-export class MeteringListComponent implements OnInit, OnChanges, AfterViewInit {
+export class MeteringReportListComponent implements OnInit {
   private readonly _unsubscribe = new Subject<void>();
   private _fetchingReport = '';
-  private _configChanged$ = new Subject<MeteringConfiguration>();
+  scheduleName: string;
+  schedule: string;
+  interval: number;
   reports: Report[] = [];
   dataSource = new MatTableDataSource<Report>();
   readonly displayedColumns: string[] = ['name', 'size', 'lastModified', 'actions'];
   @ViewChild(MatSort, {static: true}) sort: MatSort;
-  @ViewChild(MatPaginator) paginator: MatPaginator;
-  @Input() config: MeteringConfiguration;
+  @ViewChild(MatPaginator, {static: true}) paginator: MatPaginator;
 
-  private _isLoading = true;
+  private _isLoadingReports = true;
+  private _isLoadingConfig = true;
 
-  get isLoading(): boolean {
-    return this._isLoading;
+  get isLoadingReports(): boolean {
+    return this._isLoadingReports;
   }
 
-  get enabled(): boolean {
-    return this.config.enabled && !!this.config.storageSize && !!this.config.storageClassName;
+  get isLoadingConfig(): boolean {
+    return this._isLoadingConfig;
   }
 
   constructor(
     private readonly _settingsService: SettingsService,
+    private readonly _meteringService: MeteringService,
     private readonly _userService: UserService,
+    private readonly _route: ActivatedRoute,
+    private readonly _router: Router,
     @Inject(DOCUMENT) private readonly _document: Document
   ) {}
 
   ngOnInit() {
     this.dataSource.data = this.reports;
     this.dataSource.sort = this.sort;
+    this.dataSource.paginator = this.paginator;
     this.sort.active = 'lastModified';
     this.sort.direction = 'desc';
 
-    merge(this._configChanged$, of(this.config))
-      .pipe(filter(config => config.enabled))
-      .pipe(switchMap(_ => this._settingsService.reports))
+    this.scheduleName = this._route.snapshot.params.scheduleId;
+
+    this._meteringService
+      .getScheduleConfiguration(this.scheduleName)
+      .pipe(takeUntil(this._unsubscribe))
+      .subscribe(config => {
+        this.scheduleName = config.name;
+        this.schedule = config.schedule;
+        this.interval = config.interval;
+        this._isLoadingConfig = false;
+      });
+
+    this._settingsService
+      .reports(this.scheduleName)
```

Comment:
I'd add a `map` here to modify the report name before using it. I see that this `trimPrefix` logic is used both here in a separate method and inside a `reportDownload`. It would be better to keep it in a single place and simplify the usage.

Reply:
I think I know what you mean but the line you selected is not the one you meant. Shouldn't it be #126?

Reply:
I was not sure how you wanted to use `map` in pipe here. I modified report name in the subscribe method to trim report name as you suggested.

Reply:
https://www.learnrxjs.io/learn-rxjs/operators/transformation/map

It works basically the same as the `array.map`. It just runs before the subscribe.